src/interpreter/crypto.rs

```diff
@@ -18,20 +17,25 @@ impl<S> Interpreter<S> {
 
         let (a, b, c) = (a as usize, b as usize, c as usize);
 
-        let bx = b + Bytes64::LEN;
-        let cx = c + Bytes32::LEN;
+        let bx = b
+            .checked_add(Bytes64::LEN)
+            .ok_or_else(|| Bug::new(BugId::ID010, BugVariant::CheckedRegisterOverflow))?;
```

Comment:
in my testing I realized that this and subsequent similar checks are only reachable if VM_MAX_RAM is nearly MAX. In the blockchain.rs test file we did these computations before the above checks. Would we like to leave these as unreachable given sane values of VM_MAX_RAM or rearrange so that all checks can be tested?

Reply:
I think for this type of situation we should perform these checks before the register comparisons to VM_MAX_RAM. The calculations are just rearrangements of the preceding checks and could be used directly  as `bx > VM_MAX_RAM` instead of `b > VM_MAX_RAM - Bytes64::LEN`

Reply:
I agree there. We are checking some stuff that is logically guarantee to be sound. It is indeed overkill and we might want to reconsider that. Since we already agreed that safe arithmetic isn't applicable everywhere, then we might be able to be flexible here, too.

Regarding your question about `VM_MAX_RAM`; yes, you are correct. If this constant is set properly, then this panic is unreachable